SetupCrossTenantRelationshipForTargetTenant.ps1

```diff
@@ -776,10 +778,59 @@ function UserPrompt() {
     $choice=$host.ui.PromptForChoice($title, $message, $options, 1)
     if ($choice -ne 0) {
         Exit}
-    else {Main}
+    Start-Sleep 2
+    Write-Host "`nWe are verifying that you are using the latest version of the script."`n
+    Write-Host "This requires that we download the latest version of the script from GitHub to compare with your local copy."
+    Write-Host "This file will be stored on your local computer temporarily, as well as overwrite your existing script file if it is out of date."
+    $title = "Confirm: Allow for download from GitHub."
+    $message = "`nIf you are ready to begin this step, select 'Y'. `nIf you would prefer to manually download the scripts to make sure you have the latest version or change your path, select 'N'"
+    $yes = New-Object System.Management.Automation.Host.ChoiceDescription "&Yes", "Yes"
+    $no = New-Object System.Management.Automation.Host.ChoiceDescription "&No", "No"
+    $options = [System.Management.Automation.Host.ChoiceDescription[]]($yes, $no)
+    $choice=$host.ui.PromptForChoice($title, $message, $options, 1)
+    if ($choice -ne 0) {
+        Exit}
+    else {Verification}
+
 }
 
-UserPrompt
+function Verification {
+    Write-Host "`nBeginning verification steps."`n
+    Write-Host "Verifying ability to create a new organization relationship in the tenant."
+    try {
+        New-OrganizationRelationship -DomainNames contoso.onmicrosoft.com -Name Contoso -WhatIf -ErrorAction Stop

```

Comment:
pipe the output of this command to Out-Null so that it doesnt pollute the output stream.
Also can this be done with Get-OrgRel instead? Just that it seems a bit better than New-OrgRel -Whatif

Reply:
I don't think Get-OrgRel would throw the error here, it just wouldn't find one.

Also, I tried | Out-Null, it didn't work in this context :(

Reply:
This 'might' be overkill here but you could add a quick function that also validates if the user has permissions to run New-OrgRel and Enable-OrgCustomization... Example (hacked from something Jason did for me in a diff script).
If you're doing this elsewhere.. ignore me.

# determine if we have the commands to run the configuration, specifically the New and Enable
# for organization config settings
function CheckForCommands 
{
  $NewOrgRel = Get-Command "New-OrganizationRelationship" -ParameterName <pick a specific param needed for this feature> -ErrorAction:SilentlyContinue;
  if ($NewOrgRel -eq $null) {
    Write-Error ("New-OrganizationRelationship is not available with your current role assignment.");
    return $false;
  }

  $EnableOrgCustomization = Get-Command "Enable-OrganizationCustomization" -ErrorAction:SilentlyContinue;
  if ($EnableOrgCustomization-eq $null) {
    Write-Error ("Enable-OrganizationCustomization is not available with your current role assignment.");
    return $false;
  }

  return $true;
}